### PR TITLE
#373 Avatar component

### PR DESCRIPTION
**Pull request by sgraczyk**
Resolves: #373 

## Description
Implements the `Avatar` component from scratch. During the development phase I made some decisions to be confirmed by the rest of the team:
* I decided to make the API more generic. For example instead of `kind: 'person' | 'group'` I've used `shape: 'circle' : 'rounded-square'`. A similar change was for the status, as not all products are chat-focused, so I avoided the `accepts chats` name.
* Figma project does not specify the behavior of rim in case of `rounded-square`, nevertheless, I implemented it.
* I'm a bit confused on how to address the position of the status icon as the design seems to do it differently for each size and square. I tried to follow that assumption with some slight modifications.

## Storybook
- [Avatar](https://feature-373--613a8e945a5665003a05113b.chromatic.com/?path=/story/components-avatar--default)

## Checklist

**Obligatory:**

- [x] Self-review
- [ ] Unit & integration tests
- [x] Storybook cases
- [x] Design review by @kstvsko 
- [x] Functional (QA) review by @TomaszRatajczyk 

**Optional:**

- [ ] Accessibility cases (keyboard control, correct HTML markup, etc.)

### CODE REVIEW

File: packages/react-components/src/components/Avatar/Avatar.helpers.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { getContrast } from 'polished';
+
+export function getInitials(name = '', count = 2): string {
```
Comment (sgraczyk) [inline packages/react-components/src/components/Avatar/Avatar.helpers.ts:3]: At this moment it does not support diacritics or emojis.

**Comment by kstvsko** [comment]
@sgraczyk 

I've added updated avatars, also with status indicators, that are now scaled and positioned at the same place at every size. I've added also the scale for the avatars with rims. :) 

Rims: https://www.figma.com/file/2pFu80PXO5A2tfyrAGnx91/Product-Components?node-id=8167%3A31183
Status Indicators: https://www.figma.com/file/2pFu80PXO5A2tfyrAGnx91/Product-Components?node-id=736%3A8527

**Comment by MichalPaszowski** [comment]
Something not pykło here https://monosnap.com/file/1aOi1ZZ2lR00rwnWGF2JmDl9Ks13r3

**Review by MichalPaszowski — APPROVED** [review]

**Comment by sgraczyk** [comment]
> Something not pykło here https://monosnap.com/file/1aOi1ZZ2lR00rwnWGF2JmDl9Ks13r3

Should be fixed within https://github.com/livechat/design-system/pull/375/commits/99b29a141f0edf68f0946b178b36731967670c83

**Comment by sgraczyk** [comment]
> @sgraczyk
> 
> I've added updated avatars, also with status indicators, that are now scaled and positioned at the same place at every size. I've added also the scale for the avatars with rims. :)
> 
> Rims: https://www.figma.com/file/2pFu80PXO5A2tfyrAGnx91/Product-Components?node-id=8167%3A31183 Status Indicators: https://www.figma.com/file/2pFu80PXO5A2tfyrAGnx91/Product-Components?node-id=736%3A8527

Status indicators are implemented as proposed, however, I'd like to wait before introducing changes designed for rims. It looks that now the size of the image is adjusted so the rim fits into the original size of the avatar. It'll probably introduce a flickering effect, especially that the rim is a UI element that may appear and disappear when a specific condition changes. Can we move the work around rims to another issue, and accept the current state for now?

**Review by kstvsko — APPROVED** [review]

**Review by mreszke — APPROVED** [review]

File: packages/react-components/src/components/Avatar/Avatar.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    expect(container.firstChild).toHaveStyle(
+      `background-color: ${customColor}`
+    );
+  });
+
+  const extraSmallSizes: AvatarProps['size'][] = [
+    'xxxsmall',
+    'xxsmall',
+    'xsmall',
+  ];
```
Comment (MichalPaszowski) [inline packages/react-components/src/components/Avatar/Avatar.spec.tsx:48]: missing space - perhaps it should be put up top of the file?

File: packages/react-components/src/components/Avatar/Avatar.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+  );
+
+  const smallAndAboveSizes: AvatarProps['size'][] = [
+    'small',
+    'medium',
+    'large',
+    'xlarge',
+    'xxlarge',
+  ];
+  it.each(smallAndAboveSizes)(
```
Comment (MichalPaszowski) [inline packages/react-components/src/components/Avatar/Avatar.spec.tsx:69]: same as above

**Pull request merged**